fasd

```diff
@@ -314,7 +315,14 @@ EOS
       *\ $1\ *) return;;
     esac
 
-    shift; fasd --add "$@" # add all arguments except command
+    shift                       # remove command itself
+
+    # possibly too many elements:
+    if [ $# -gt $_FASD_ADD_MAX ]; then
+        shift $(( $# - _FASD_ADD_MAX ))
```

Comment:
FWIW I found it surprising that this removed the *first* N excess arguments rather than the *last* N arguments.  That said, I don't know a better way to do this in POSIX shell that isn't some big nasty loop.  (I hit the same problem this PR tries to address.  I'm now trying to migrate away from fasd to something else.)